packages/ember-data/lib/system/model/model.js

```diff
@@ -781,6 +781,25 @@ var Model = Ember.Object.extend(Ember.Evented, {
 
   hasMany: function() {
     Ember.assert("The `hasMany` method is not available on DS.Model, a DS.Snapshot was probably expected. Are you passing a DS.Model instead of a DS.Snapshot to your serializer?", false);
+  },
+
+  // handlers for the id property
+  unknownProperty: function (key) {
```

Comment:
Well, this unknown property only needs to be in place until ember 2.2, as I think the object.defineProperty way should work then via rwjblue's fix